roles/tests/test_user_models.py

```diff
@@ -51,3 +51,11 @@ def __init__(self, foo_var):
 
         user_group2 = FooObject('bar')
         self.assertFalse(user_group1.__eq__(user_group2))
+
+    def test_user_group_hash(self):
+        """Test for UserGroup __hash__"""
+        user_group = UserGroup('foo1', 'bar')
+        self.assertEqual(
+            user_group.__hash__(),
+            hash((user_group.username, user_group.group_type))
```

Comment:
You still haven't convinced me :-) :
since the `__hash__` function is defined as `hash` of a tuple and since tuple are hashable in python, testing that changing a value in the tuple gives a different hash, to me it means testing the behavior of the `hash` function.

But in any case, it's no big deal: I can add the tests.